### ISSUE TITLE
Adaptive Wavelet Network-Based Predictive Control for a Collaborative Manipulator Arm in Industry 4.0

### ISSUE THREAD

**Issue by wadhahtoumi08** [marp-team/marp-vscode#535]
---
marp: true
theme: industry4
paginate: true
footer: 'Wadhah Toumi | WNN-MPC for Collaborative Manipulators | Industry 4.0 | 2025'
style: |
  :root {
    --color-primary: #2c3e50;
    --color-secondary: #3498db;
    --color-highlight: #e74c3c;
    --color-background: #ffffff;
    --color-foreground: #2c3e50;
    --font-primary: 'Helvetica Neue', Arial, sans-serif;
    --font-secondary: 'Times New Roman', serif;
  }

  section {
    font-family: var(--font-primary);
    color: var(--color-foreground);
    background-color: var(--color-background);
  }

  h1, h2, h3 {
    color: var(--color-primary);
    font-family: var(--font-primary);
    font-weight: 600;
  }

  h1 {
    font-size: 2.5rem;
    text-align: center;
    margin-bottom: 0.5em;
  }

  h2 {
    font-size: 2rem;
    color: var(--color-secondary);
    border-bottom: 2px solid var(--color-secondary);
    padding-bottom: 0.2em;
  }

  .highlight {
    color: var(--color-highlight);
    font-weight: bold;
  }

  .data {
    background-color: #f8f9fa;
    padding: 8px 12px;
    border-radius: 4px;
    border-left: 4px solid var(--color-secondary);
    margin: 10px 0;
    font-family: var(--font-secondary);
  }

  table {
    width: 100%;
    border-collapse: collapse;
    margin: 20px 0;
    font-family: var(--font-secondary);
  }

  th, td {
    border: 1px solid #ddd;
    padding: 10px;
    text-align: center;
  }

  th {
    background-color: var(--color-primary);
    color: white;
  }

  img[alt~="center"] {
    display: block;
    margin: 0 auto;
  }

  .two-column {
    display: grid;
    grid-template-columns: 1fr 1fr;
    gap: 20px;
  }

  .three-column {
    display: grid;
    grid-template-columns: 1fr 1fr 1fr;
    gap: 15px;
  }

  .math {
    font-family: var(--font-secondary);
    font-size: 1.1rem;
    color: var(--color-primary);
    background-color: #f8f9fa;
    padding: 10px;
    border-radius: 4px;
    text-align: center;
    margin: 15px 0;
  }

  footer {
    font-size: 0.6rem;
    color: #7f8c8d;
    text-align: center;
  }

---

# **Adaptive Wavelet Network-Based Predictive Control**
## **For Collaborative Manipulator Arms in Industry 4.0**

**Wadhah Toumi**
*Research Lab MACS, ENIG | University of Gabes*
**Hatem Tlijani** | **Assem Thabet**

![center w:400](https://via.placeholder.com/800x200/2c3e50/ffffff?text=ENIG+Logo+|+Industry+4.0) 

---

# **Context: Industry 4.0 and Collaborative Robots**

*"How to achieve **precision**, **safety**, and **adaptability** in dynamic industrial environments with human-robot collaboration?"*

![bg right:40% w:500](https://via.placeholder.com/500x400/2c3e50/ffffff?text=Figure+1) 

---
## **Control Challenges for Cobots**

<div class="two-column">

**Technical Challenges**
- **Nonlinear dynamics** (friction, variable inertia)
- **Parametric uncertainties** (load variations, mechanical degradation)
- **External disturbances** (human operator, environment)
- **Strict safety constraints** (ISO 10218, ISO/TS 15066 standards)

|||

**Limitations of Existing Methods**
- **PID**: Degraded performance outside nominal points
- **Classical MPC**: Requires precise system model
- **Standard Neural Networks**: Slow convergence, initialization sensitivity

</div>

---
# **Our Solution: Hybrid WNN-MPC Architecture**

![center w:600](https://via.placeholder.com/800x400/2c3e50/ffffff?text=Figure+5) 

- **Wavelet Neural Network (WNN)**: Adaptive approximation of uncertain dynamics
- **Model Predictive Control (MPC)**: Constraint-aware optimization with safety terms
- **Online adaptation**: No need for precise system model

---
## **Wavelet Neural Network (WNN)**

<div class="math">

**Mexican Hat Wavelet Function:**
$$
\psi(x) = (1 - x^2) e^{-x^2/2}
$$

**Network Output:**
$$
f(x) \approx \sum_{i=1}^N w_i \psi\left(\frac{x - b_i}{a_i}\right)
$$

</div>

<div class="two-column">

**3-Layer Architecture**
- Better **expressiveness** for nonlinearities
- **Time-frequency localization** (superior to classical networks)
- **Fast convergence** through adaptive parameters $a_i$, $b_i$

|||

**Advantages vs. 2-Layer**

| Feature               | 2-Layer        | 3-Layer (Our Approach) |
|-----------------------|----------------|------------------------|
| Coupled nonlinearities | Limited        | **Excellent**          |
| Variable inertia       | Poor           | **Good**               |
| Disturbance adaptation | Poor           | **Excellent**          |

</div>

---
## **Enhanced Model Predictive Control**

<div class="math">

**Cost Function with Safety Constraints:**
$$
J = \sum_{k=1}^{N_p} \|q_{ref,k} - q_k\|_Q^2 + \sum_{k=0}^{N_c-1} \left( \|\Delta \tau_k\|_R^2 + \|\tau_k - \tau_{safe,k}\|_S^2 \right)
$$

- $N_p = 20$ (prediction horizon), $N_c = 10$ (control horizon)
- $\tau_{safe,k}$: Safe control input from human-robot distance
- **Optimization**: Solved via **Sequential Quadratic Programming (SQP)**

</div>

---
# **Experimental Results**

## **Simulation Setup**
- **Manipulator**: 2-DOF planar (links: 1m, masses: 1kg)
- **WNN**: 10 wavelet functions, $\eta = 0.01$
- **MPC**: $Q = \text{diag}(10,10)$, $R = \text{diag}(0.1,0.1)$, $S = \text{diag}(100,100)$
- **Test Scenarios**:
  1. Nominal operation (circular trajectory)
  2. Human interaction (disturbances at 5-7s)
  3. Load variation (+30% at t=3s)
  4. Random external disturbances

---
## **Performance Comparison**

<div class="data">

| Metric                  | Classical MPC | **WNN-MPC**       | Improvement |
|-------------------------|----------------|-------------------|--------------|
| **Tracking Accuracy (RMSE)** | 0.038 rad   | **0.025 rad**     | **35%**      |
| **Control Effort**      | 12.5 Nm        | **7.5 Nm**        | **40%**      |
| **Response Time**       | 1.8 s          | **1.2 s**         | **33%**      |
| **Robustness (Std Dev)** | 0.042 rad      | **0.018 rad**     | **57%**      |

</div>

---

### **Trajectory Tracking**
![center w:250](https://via.placeholder.com/300x200/2c3e50/ffffff?text=Figure+6) 
- Precise tracking even with disturbances

|||
### **Tracking Error**
![center w:250](https://via.placeholder.com/300x200/2c3e50/ffffff?text=Figure+7) 
- Error spike controlled during human interaction

|||
### **Control Effort**
![center w:250](https://via.placeholder.com/300x200/2c3e50/ffffff?text=Figure+9) 
- **40% reduction** = energy savings

---
# **Scenario Analysis**

<div class="two-column">

**1. Human Interaction (5-7s)**
- Simulated external forces
- **57.1% reduction** in error standard deviation
- **Stability maintained** despite disturbances

|||

**2. Load Variation (+30%)**
- Sudden load change at t=3s
- **75% faster settling time** (0.5s vs 2.0s)
- **66.7% less overshoot** (5% vs 15%)

</div>

---
# **Why Choose WNN-MPC?**

<div style="display: grid; grid-template-columns: 1fr 1fr; gap: 20px;">

**Key Advantages**
- **Adaptive**: No precise model required
- **Safe**: Real-time constraint satisfaction
- **Efficient**: 40% control effort reduction
- **Robust**: Maintains performance under disturbances

|||

**Potential Applications**
- **Industry 4.0**: Flexible production lines
- **Medical**: Surgical assistance robots
- **Logistics**: Automated warehouse manipulators
- **Aerospace**: Robotic arms for maintenance

</div>

---
# **Future Work**

1. **Experimental Validation**:
   - Tests on real cobots (e.g., UR5, KUKA LBR arms)
2. **Digital Twin Integration**:
   - Hybrid physical-virtual simulation
3. **Multi-Robot Collaboration**:
   - Extension to multi-agent systems
4. **Reinforcement Learning**:
   - Online optimization of controller parameters

---

# **Conclusion**

### *"A significant advancement for **smarter, safer, and more efficient** cobots in Industry 4.0."*

**Key Results**:
✅ **35% better tracking accuracy** | ✅ **40% energy savings**
✅ Robust to disturbances | ✅ Real-time adaptability

**Next Steps**:
- Industrial partner collaborations for real-world testing
- Integration with IoT/Cloud platforms (e.g., Siemens MindSphere)

---

# **Thank You!**
**Questions?**

**Contact**: wadhahtoumi08@gmail.com
**Lab**: Research Lab MACS, ENIG | University of Gabes

![center w:300](https://via.placeholder.com/600x200/2c3e50/ffffff?text=Thank+You!+|+Industry+4.0) 

---
# **Appendix: Key References**

1. **Wavelet Neural Networks**:
   - [Quynh et al., 2022] - WNN for robotic manipulators
   - [Chen et al., 2022] - Adaptive WNN control

2. **Model Predictive Control**:
   - [Kang et al., 2022] - Neural network-based MPC
   - [Limon et al., 2020] - MPC for nonlinear systems

3. **Cobots & Industry 4.0**:
   - [Matheson et al., 2020] - Human-robot collaboration
   - [Singh et al., 2021] - Digital twins in Industry 4.0

---
# **Technical Parameters (Appendix)**

```markdown
**2-DOF Manipulator**:
- Link lengths: l₁ = l₂ = 1.0 m
- Masses: m₁ = m₂ = 1.0 kg
- Viscous friction: b₁ = b₂ = 0.1 Nm/s

**WNN Configuration**:
- Wavelet functions: N = 10
- Type: Mexican hat wavelet
- Learning rate: η = 0.01

**MPC Settings**:
- Sampling period: Tₛ = 0.01 s
- Horizons: Nₚ = 20, N_c = 10
- Weight matrices: Q = diag(10,10), R = diag(0.1,0.1), S = diag(100,100)
